CMakeLists.txt

```diff
@@ -47,6 +47,8 @@ OPTION(XEUS_CPP_BUILD_EXECUTABLE "Build the xcpp executable" ON)
 OPTION(XEUS_CPP_USE_SHARED_XEUS "Link xcpp  with the xeus shared library (instead of the static library)" ON)
 OPTION(XEUS_CPP_USE_SHARED_XEUS_CPP "Link xcpp  with the xeus shared library (instead of the static library)" ON)
 
+option(CMAKE_COMPILE_WARNING_AS_ERROR "Compile warnings as errors" ON)
```

Comment:
- No need to create an option here, `CMAKE_COMPILE_WARNING_AS_ERROR` already exists as a [cmake variable](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_WARNING_AS_ERROR.html) and can be passed directly in the command line.
- Although it's a good thing to have it enabled by default on the CI, it should be off by default for other users
- The required minimal version of cmake should be updated to be compatible with the use of this variable.

Reply:
This comment has now been addressed.